darwin/utils/utils.py

```diff
@@ -693,6 +705,74 @@ def _parse_darwin_video_annotation(annotation: dict) -> Optional[dt.VideoAnnotat
     return main_annotation
 
 
+def _parse_darwin_raster_annotation(annotation: dict) -> Optional[dt.Annotation]:
+    if not annotation.get("raster_layer"):
+        raise ValueError("Raster annotation must have a 'raster_layer' field")
+
+    id: Optional[str] = annotation.get("id")
+    name: Optional[str] = annotation.get("name")
+    raster_layer: Optional[dt.JSONFreeForm] = annotation.get("raster_layer")
+    slot_names: Optional[List[str]] = parse_slot_names(annotation)
+
+    if not id or not name or not raster_layer or not slot_names:
+        raise ValueError("Raster annotation must have an 'id', 'name', 'slot_names' and 'raster_layer' field")
+
+    dense_rle, mask_annotation_ids_mapping, total_pixels = (
+        raster_layer.get("dense_rle", None),
+        raster_layer.get("mask_annotation_ids_mapping", None),
+        raster_layer.get("total_pixels", None),
+    )
+
+    if not dense_rle or not mask_annotation_ids_mapping or not total_pixels:
+        raise ValueError(
+            "Raster annotation must have a 'dense_rle', 'mask_annotation_ids_mapping' and 'total_pixels' field"
+        )
+
+    new_annotation = dt.Annotation(
+        dt.AnnotationClass(name, "raster_layer"),
+        {
+            "id": id,
+            "name": name,
+            "raster_layer": {
+                "dense_rle": dense_rle,
+                "mask_annotation_ids_mapping": mask_annotation_ids_mapping,
+                "total_pixels": total_pixels,
+            },
+        },
+        slot_names=slot_names,
+    )
+
+    return new_annotation
+
+
+def _parse_darwin_mask_annotation(annotation: dict) -> Optional[dt.Annotation]:
+    if not annotation.get("mask"):
+        raise ValueError("Mask annotation must have a 'mask' field")
+
+    id: Optional[str] = annotation.get("id")
+    name: Optional[str] = annotation.get("name")
+    mask: Optional[dt.JSONFreeForm] = annotation.get("mask")
+    slot_names: Optional[List[str]] = parse_slot_names(annotation)
+
+    if not id or not name or not mask or not slot_names:
```

Comment:
As above, when we have a pydantic basemodel approach, this sort of check can be replaced with on creation validation

Reply:
Yep, and noted again.  Feel free to ticket.